native/src/utils/calendarRangeUtils.ts

```diff
@@ -0,0 +1,61 @@
+import { DateTime } from 'luxon'
+
+import { lightColors } from 'build-configs/integreat/theme/colors'
+
+type MarkedDateType = {
+  selected: boolean
+  startingDay?: boolean
+  endingDay?: boolean
+}
+
+export const getMarkedDates = (
+  startDate: DateTime | null,
+  endDate: DateTime | null,
+): Record<string, MarkedDateType> => {
+  const markedDateStyling = {
+    color: lightColors.themeColor,
+    textColor: lightColors.textColor,
+  }
+
+  const markedDates: Record<string, MarkedDateType> = {}
+
+  const cutoffDate = DateTime.fromISO('2020-01-01')
+
+  const validStartDate = startDate && startDate >= cutoffDate ? startDate : null
+
+  const validEndDate = endDate && endDate >= cutoffDate ? endDate : null
+
+  if (validStartDate) {
+    const startDateString = validStartDate.toISODate()
+    markedDates[startDateString] = {
+      selected: true,
+      startingDay: true,
+      endingDay: startDateString === validEndDate?.toISODate(),
+      ...markedDateStyling,
+    }
+  }
+
+  if (validEndDate && (!validStartDate || validStartDate.toISODate() !== validEndDate.toISODate())) {
+    const endDateString = validEndDate.toISODate()
+    markedDates[endDateString] = {
+      selected: true,
+      endingDay: true,
+      ...markedDateStyling,
+    }
+  }
+
+  if (validStartDate && validEndDate && validStartDate < validEndDate) {
+    let current = validStartDate.plus({ days: 1 })
+
+    while (current < validEndDate) {
+      const dateString = current.toISODate()
+      markedDates[dateString] = {
+        selected: true,
+        ...markedDateStyling,
+      }
+      current = current.plus({ days: 1 })
+    }
+  }
+
+  return markedDates
```

Comment:
I think this could be simplified quite a bit, I think the following should do the same:
```suggestion
   const cutoffDate = DateTime.fromISO('2020-01-01')

  if (!startDate || startDate < cutoffDate) {
    return {}
  }

  const safeEndDate = endDate ?? startDate
  const markedDates: Record<string, MarkedDateType> = {}

  let currentDate = startDate
  while (currentDate <= safeEndDate) {
    markedDates[currentDate.toISODate()] = {
      selected: true,
      startingDay: currentDate === startDate,
      endingDay: currentDate === endDate,
      ...markedDateStyling,
    }
    currentDate = currentDate.plus({ days: 1 })
  }
  
  return markedDates
```

Reply:
this will miss up the marking for end date and when you click the same date twice..
![image](https://github.com/user-attachments/assets/e408dc69-9d28-4b57-b14f-04b711df2171)

Reply:
I adjusted the suggestion yesterday and replaced `currentDate === endDate` with `endDate?.equals(currentDate)`, that should do the trick to include marking for the last day too.